### PR TITLE
v4.1.1

### PR DESCRIPTION
**Pull request by WalshyDev**
## 4.1 release
FlareBot v4.1.1 - Internal changes mostly

### Changed
* All the getXByID to getXById
* Moved over to shardManager and no longer use the JDA[]
* Moved the Player listener to its own class, now it's a lot less messy and it will be easier for us to work on in the future.
* Increased ShardUtils#isDead timeout so that we can get a more accurate count
* Cleaned up the code a bit, always good!

### Fixed
* Fixed `forceban` showing `ban` in the usage
* Fixed internal commands being default

### CODE REVIEW

File: src/main/java/stream/flarebot/flarebot/Events.java [hunk 1]
```diff
@@ -102,7 +102,8 @@ public void onMessageReactionAdd(MessageReactionAddEvent event) {
 
     @Override
     public void onReady(ReadyEvent event) {
-        flareBot.latch.countDown();
+        if(FlareBot.getInstance().isReady())
```
Comment (weeryan17) [inline src/main/java/stream/flarebot/flarebot/Events.java]: 👀

File: src/main/java/stream/flarebot/flarebot/FlareBot.java [hunk 2]
```diff
@@ -250,118 +237,29 @@ public static Prefixes getPrefixes() {
...
-                clients[0] = builder.buildBlocking(JDA.Status.AWAITING_LOGIN_CONFIRMATION);
-                Thread.sleep(5000);
-            } else {
-                builder = builder.setReconnectQueue(new SessionReconnectQueue());
-                for (int i = 0; i < clients.length; i++) {
-                    clients[i] = builder.useSharding(i, clients.length).buildBlocking(JDA.Status.AWAITING_LOGIN_CONFIRMATION);
-                    Thread.sleep(5000); // 5 second backoff
-                }
-            }
+                    .setShardsTotal(2)
```
Comment (weeryan17) [inline src/main/java/stream/flarebot/flarebot/FlareBot.java]: ummmmmmmm

File: src/main/java/stream/flarebot/flarebot/commands/general/StatusCommand.java [hunk 3]
```diff
@@ -19,17 +19,17 @@
...
 
         int deadShard = 0;
         int reconnecting = 0;
         int connecting = 0;
         int noVoiceConnections = 0;
         int highResponseTime = 0;
 
-        for (int shardId = 0; shardId < fb.getClients().length; shardId++) {
-            JDA jda = fb.getClients()[shardId];
+        for(int shardId = 0; shardId < fb.getShards().size(); shardId++) {
```
Comment (weeryan17) [inline src/main/java/stream/flarebot/flarebot/commands/general/StatusCommand.java]: 👀

File: src/main/java/stream/flarebot/flarebot/Events.java (hunk 1 above)
Comment (WalshyDev) [reply src/main/java/stream/flarebot/flarebot/Events.java]: Problem?

Comment (weeryan17) [reply src/main/java/stream/flarebot/flarebot/Events.java]: formatting

File: src/main/java/stream/flarebot/flarebot/FlareBot.java [hunk 4]
```diff
@@ -250,118 +237,29 @@ public static Prefixes getPrefixes() {
...
-                clients[0] = builder.buildBlocking(JDA.Status.AWAITING_LOGIN_CONFIRMATION);
-                Thread.sleep(5000);
-            } else {
-                builder = builder.setReconnectQueue(new SessionReconnectQueue());
-                for (int i = 0; i < clients.length; i++) {
-                    clients[i] = builder.useSharding(i, clients.length).buildBlocking(JDA.Status.AWAITING_LOGIN_CONFIRMATION);
-                    Thread.sleep(5000); // 5 second backoff
-                }
-            }
+                    .setShardsTotal(-1)
```
Comment (binaryoverload) [inline src/main/java/stream/flarebot/flarebot/FlareBot.java]: This isn't needed, it's the default behaviour of the builder

File: src/main/java/stream/flarebot/flarebot/commands/moderation/SetPrefixCommand.java [hunk 5]
```diff
@@ -28,7 +28,6 @@ public void onCommand(User sender, GuildWrapper guild, TextChannel channel, Mess
                         .queue();
             } else {
                 MessageUtils.sendErrorMessage("Cannot set the prefix to be more that one character long!", channel, sender);
-                return;
```
Comment (binaryoverload) [inline src/main/java/stream/flarebot/flarebot/commands/moderation/SetPrefixCommand.java]: Why was this removed?

File: src/main/java/stream/flarebot/flarebot/commands/music/SongNickCommand.java [hunk 6]
```diff
@@ -21,7 +21,6 @@ public void onCommand(User sender, GuildWrapper guild, TextChannel channel, Mess
             if (GeneralUtils.canChangeNick(guild.getGuildId()))
                 channel.getGuild().getController().setNickname(channel.getGuild().getSelfMember(), null).queue();
             MessageUtils.sendSuccessMessage("Disabled changing nickname with song!", channel, sender);
-            return;
```
Comment (binaryoverload) [inline src/main/java/stream/flarebot/flarebot/commands/music/SongNickCommand.java]: Same, these stop the usage firing IIRC

File: src/main/java/stream/flarebot/flarebot/commands/music/SongNickCommand.java [hunk 7]
```diff
@@ -43,7 +42,6 @@ public void onCommand(User sender, GuildWrapper guild, TextChannel channel, Mess
                         .queue();
             }
             MessageUtils.sendSuccessMessage("Enabled changing nickname with song!", channel, sender);
-            return;
```
Comment (binaryoverload) [inline src/main/java/stream/flarebot/flarebot/commands/music/SongNickCommand.java]: Same

File: src/main/java/stream/flarebot/flarebot/FlareBot.java (hunk 4 above)
Comment (WalshyDev) [reply src/main/java/stream/flarebot/flarebot/FlareBot.java]: Possibly, but it still makes it easier to switch to like 2 shards when we want to test multi-shard things

File: src/main/java/stream/flarebot/flarebot/commands/moderation/SetPrefixCommand.java (hunk 5 above)
Comment (WalshyDev) [reply src/main/java/stream/flarebot/flarebot/commands/moderation/SetPrefixCommand.java]: Prefix has no usage message, everything is handled inside the command.

File: src/main/java/stream/flarebot/flarebot/commands/music/SongNickCommand.java (hunk 6 above)
Comment (WalshyDev) [reply src/main/java/stream/flarebot/flarebot/commands/music/SongNickCommand.java]: It's a toggle, no usage.

File: src/main/java/stream/flarebot/flarebot/commands/music/SongNickCommand.java (hunk 7 above)
Comment (WalshyDev) [reply src/main/java/stream/flarebot/flarebot/commands/music/SongNickCommand.java]: Same as above.
